Sources/SpeziHealthCharts/MetricChart/ChartViews/HealthChart.swift

```diff
@@ -0,0 +1,60 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+import HealthKit
+
+
+public struct HealthChart: View {
+    @State private var privateRange: ChartRange
+    private var privateRangeBinding: Binding<ChartRange>?
+    
+    var range: Binding<ChartRange> {
+        Binding(
+            get: {
+                privateRangeBinding?.wrappedValue ?? privateRange
+            }, set: { newValue in
+                if let privateRangeBinding {
+                    privateRangeBinding.wrappedValue = newValue
+                } else {
+                    privateRange = newValue
+                }
+            }
+        )
+    }
+    
+    private let quantityType: HKQuantityType
+    private let dataProvider: any DataProvider
+    
+    
+    public var body: some View {
+        InternalHealthChart(quantityType, range: range, provider: dataProvider)
+    }
+    
+    
+    public init(
+        _ type: HKQuantityType,
+        in initialRange: ChartRange = .month,
+        provider: any DataProvider = HealthKitDataProvider()
```

Comment:
What was the reason to have the concept of the `DataProvider` be part of the public interface here? Would it make sense to just to just pass a collection of (HK) samples to the chart to be as reusable as possible. For example, if I just have a few recordings form my Bluetooth device that I turned into HKQuantitySamples, I can just pass the collection to the Chart.
I think the concept of the `HealthKitDataProvider` API is great, especially to make it easier to query samples from the HealthKitStore. But can these two APIs be separate? Or have a chart view that works with a DataProvider be a separate component that sits on top of a simple chart implementation that just takes a collection of samples?
Let me know what your thoughts are here 👍